model/helpers.go

```diff
@@ -5,15 +5,12 @@
 package model
 
 import (
-	"time"
-
 	cloudModel "github.com/mattermost/mattermost-cloud/model"
 )
 
-// Timestamp produces a millisecond-precision timestamp in a standard
-// way with the other Mattermost APIs
-func Timestamp() int64 {
-	return time.Now().UnixNano() / 1000
+// GetMillis is a convenience method to get milliseconds since epoch.
+func GetMillis() int64 {
+	return cloudModel.GetMillis()
```

Comment:
As you mentioned previous logic was not actually calculating milliseconds and I think it is good that we correct it. However, as there are already existing records in DBs I would be 3/5 on running DB migration to correct those. Because we use tombstones instead of full delete the only way to get rid of old records would be to wipe them manually and in six months we might forget about it and we will be looking for a bug why some timestamps are in the future 😛

Reply:
Yep, fair point! I will add the migration.